pkg/utils/utils_virtual.go

```diff
@@ -118,6 +119,27 @@ func GetOpenstackData(useHostPath bool) (metaData *OSPMetaData, networkData *OSP
 	if err != nil {
 		metaData, networkData, err = getOpenstackDataFromMetadataService()
 	}
+
+	// We can't rely on the PCI address from the metadata so we will lookup the real PCI address
+	// for the NIC that matches the MAC address.
+	//
+	// Libvirt/QEMU cannot guarantee that the address specified in the XML will match the address seen by the guest.
+	// This is a well known limitation: https://libvirt.org/pci-addresses.html
+	// When using the q35 machine type, it highlights this issue due to the change from using PCI to PCI-E bus for virtual devices.
+	//
+	// With that said, the PCI value in Nova Metadata is a best effort hint due to the limitations mentioned above. Therefore
+	// we will lookup the real PCI address for the NIC that matches the MAC address.
+	for i, device := range metaData.Devices {
+		realPCIAddr, err := getPCIAddressFromMACAddress(device.Mac)
+		if err != nil {
+			return metaData, networkData, fmt.Errorf("GetOpenstackData(): error getting PCI address for device %s: %w", device.Mac, err)
```

Comment:
agree with @adrianchiris in case the SNO file is deleted and we restart the daemon AND there is a workload running using the device we will not find it so we should not return and error in case it was not found (let's put a warning)

A better solution we can implement in case:
1. SNO doesn't exist
2. daemon is restarted
3. we have sriov allocated devices

is to drain the node so will remove the devices but it's a pretty big change in the code today so it can be implemented in a separate PR because it's not related to this fix, it's a bug that exist even before this.

here we just need to add a warning if the function `getPCIAddressFromMACAddress` returns "device not found base on mac" and not return error that will kill the pod

Reply:
Good for me Seb, I'm adding a comment as well about this plan.